packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -150,12 +151,22 @@ function GridColumnHeaderItem(props: GridColumnHeaderItemProps) {
     ariaSort = sortDirection === 'asc' ? 'ascending' : 'descending';
   }
 
+  React.useEffect(() => {
+    if (!showColumnMenuIcon) {
+      setShowColumnMenuIcon(columnMenuOpen);
```

Comment:
To only trigger this effect when the `columnMenuOpen` prop changes, should we remove this `!showColumnMenuIcon`
Updating from `true` to `true` is not a problem no ?

Reply:
The purpose of this effect is: `showColumnMenuIcon` should always reflect the value of `columnMenuOpen` UNLESS `showColumnMenuIcon=true`.

It could be rewrote as `setShowColumnMenuIcon(value => !value ? columnMenuOpen : value);`. By the time this effect run, `columnMenuOpen` will already be false.

> Updating from true to true is not a problem no ?

No, React won't rerender the component if the value didn't change.